tsconfig.json

```diff
@@ -0,0 +1,25 @@
+{
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
+    "declaration": true,
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "jsx": "react",
+    "lib": ["dom", "es2019"],
+    "isolatedModules": true,
+    "module": "esnext",
```

Comment:
[curious] Was these config options largely copy/pasted from `@edx/paragon`'s example (looks like the config is identical)?

If so, I think we need to change the `tsconfig.json` file in `@edx/paragon`, too haha.

Reason being, I don't think we should be setting `module` or `target` to be "esnext":

> The special ESNext value refers to the highest version your version of TypeScript supports. This setting should be used with caution, since it doesn’t mean the same thing between different TypeScript versions and can make upgrades less predictable ([docs](https://www.typescriptlang.org/tsconfig#target))

While Open edX no longer needs to transpile down to ES5 to supporting legacy old browsers (e.g., IE11), we should be transpiling down to ES6. I'm also going to file an issue to this effect in the `@edx/paragon` repository.

Reply:
It was originally copy-pasted from @edx/paragon.  I've updated the references to point to the example folder.